### ISSUE TITLE
BFG Repo Cleaner Extension

### ISSUE THREAD

**Issue by Bluscream** [gitextensions/gitextensions#11336]
### Feature description

It would be nice to have a GUI of https://rtyley.github.io/bfg-repo-cleaner/ either built-in or as a plugin!

### Environment

- Git Extensions 4.1.0.16698
- Build 1fe52a137acef1fbd2b653dea263b44ad5844177
- Git 2.42.0.windows.2
- Microsoft Windows NT 10.0.22621.0
- .NET 6.0.24
- DPI 96dpi (no scaling)
- Portable: False
- Microsoft.WindowsDesktop.App Versions

```
    Microsoft.WindowsDesktop.App 3.0.3 [c:\PROGRA~1\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 3.1.32 [c:\PROGRA~1\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 5.0.17 [c:\PROGRA~1\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 6.0.24 [c:\PROGRA~1\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 7.0.13 [c:\PROGRA~1\dotnet\shared\Microsoft.WindowsDesktop.App]
    Microsoft.WindowsDesktop.App 8.0.0-rc.2.23479.10 [c:\PROGRA~1\dotnet\shared\Microsoft.WindowsDesktop.App]
```

**Comment by pmiossec** [comment]
Not sure it is a good idea as the recommend tool seems to be [git filter-repo](https://github.com/newren/git-filter-repo) (which seems to be endorsed by git team).

**Comment by Bluscream** [comment]
> Not sure it is a good idea as the recommend tool seems to be [git filter-repo](https://github.com/newren/git-filter-repo) (which seems to be endorsed by git team).

You're right, git filter-repo is more powerful, and i mean if in the end there's a nice, easy to use GUI ontop then the end-user probably doesn't care what is used in the background. I would say the plugin doesn't even need to support **everything** that filter-repo has to offer anyway, since there's really only two things that happen often enough to warrant a UI (pruning sensitive info like passwords) and accidentally commited files that became too large (>100MB in case of github lfs)
